pkg/cmd/codespace/create.go

```diff
@@ -118,7 +120,15 @@ func (a *App) Create(ctx context.Context, opts createOptions) error {
 		Location:   opts.location,
 	}
 
-	promptForRepoAndBranch := userInputs.Repository == ""
+	if opts.useWeb && (opts.displayName != "" || opts.idleTimeout != 0 || opts.retentionPeriod.Duration != nil) {
+		return errors.New("using --web with --display-name, --idle-timeout, or --retention-period is not yet supported")
```

Comment:
Nit: I don't think we want to say `yet` as it sounds like we plan to do it later, but that's not really the case right now. 

```suggestion
        return errors.New("using --web with --display-name, --idle-timeout, or --retention-period is not supported")
```